src/test/java/org/gaul/s3proxy/AwsSdkTest.java

```diff
@@ -363,11 +365,26 @@ public void testAwsV2UrlSigningWithOverrideParameters() throws Exception {
         headerOverride.setContentType("text/plain");
         generatePresignedUrlRequest.setResponseHeaders(headerOverride);
 
-        Date expiration = new Date(System.currentTimeMillis() +
-                TimeUnit.HOURS.toMillis(1));
+        Date expiration = new Date((System.currentTimeMillis() +
+                TimeUnit.HOURS.toMillis(1)) / 1000 * 1000);
         generatePresignedUrlRequest.setExpiration(expiration);
 
         URL url = client.generatePresignedUrl(generatePresignedUrlRequest);
+        List<NameValuePair> params = URLEncodedUtils.parse(url.toURI(),
+                "UTF-8");
+        for (NameValuePair param : params) {
+            if (param.getName().equals("response-content-disposition")) {
+                assertThat(param.getValue())
+                        .isEqualTo(headerOverride.getContentDisposition());
+            } else if (param.getName().equals("response-content-type")) {
+                assertThat(param.getValue())
+                        .isEqualTo(headerOverride.getContentType());
```

Comment:
Instead of testing the request parameters, you can test the response parameters if you use `url.getConnection` instead of `url.openStream`.